### PR TITLE
Fix linear pool composition balances

### PR DESCRIPTION
**Pull request by evgenyboxer**
# Description

Now that `totalSupply` is computed correctly its possible to get the right balances. Awaiting on Markus to confirm if the numbers look right.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other

## How should this be tested?

Please provide instructions so we can test. Please also list any relevant details for your test configuration.

- [ ] Check pool composition

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code where relevant, particularly in hard-to-understand areas
- [ ] My changes generate no new console warnings
- [ ] The base of this PR is `master` if hotfix, `develop` if not

### CODE REVIEW

File: src/components/contextual/pages/pool/PoolBalancesCard/BoostedPool.vue [hunk 1]
```diff
@@ -39,6 +41,17 @@ function getUnderlyingTokens(address: string) {
     ? [linearPools[address].mainToken, linearPools[address].wrappedToken]
     : [];
 }
+
+function getTokenShare(address: string) {
+  const totalSupply = props.pool.onchain.totalSupply;
```
Comment (evgenyboxer) [inline src/components/contextual/pages/pool/PoolBalancesCard/BoostedPool.vue:46]: I suppose this total supply applies to all the linear pools (bb-a-USDT/AUSDC/aDAI)

**Review by garethfuller — COMMENTED** [review]
@evgenyboxer I don't think this change is required? If you add up the USD values in the base branch it matches the pool value. If you add up the balances in this branch it's wildly off. Not sure why you thought this change was necessary?

File: src/components/contextual/pages/pool/PoolBalancesCard/BoostedPool.vue [hunk 2]
```diff
@@ -39,6 +41,17 @@ function getUnderlyingTokens(address: string) {
...
+  const token = props.pool.onchain.tokens[address];
+
+  return token != null
+    ? bnum(token.balance)
+        .div(totalSupply)
```
Comment (markusbkoch) [inline src/components/contextual/pages/pool/PoolBalancesCard/BoostedPool.vue:51]: This would have to be the `totalSupply` of `token`, not of the pool.

**Comment by markusbkoch** [comment]
> @evgenyboxer I don't think this change is required? If you add up the USD values in the base branch it matches the pool value. If you add up the balances in this branch it's wildly off. Not sure why you thought this change was necessary?

Ev was trying to address an issue that we found when checking the balances of the first stable phantom pool deployed to Kovan. The frontend assumes that all of the BPT of the linear pools are held in the stable phantom pool, but that assumption might not always hold.

Since we're no longe tracking that first kovan pool from the old factory on the subgraph, I used balpy to withdraw `bb-a-USDT` from the `bb-a-USD` pool so that we can test this.

Currently the [virtualSupply](https://kovan.etherscan.io/address/0xe667d48618e71c2a02e4a1b66ed9def1426938b6#readContract) of bb-a-USDT is 1,020,668
I hold [9,001 bb-a-USDT](https://kovan.etherscan.io/token/0xe667d48618e71c2a02e4a1b66ed9def1426938b6#balances)
The bb-a-USD pool holds 1,011,667 bb-a-USDT ([getPoolTokens, 0x8fd162f338b770f7e879030830cde9173367f3010000000000000000000004d8](https://kovan.etherscan.io/address/0xba12222222228d8ba445958a75a0704d566bf2c8#readContract))

So the pool holds 99% of the supply of bb-a-USDT.

The bb-a-USDT pool in turn holds ([getPoolTokens, 0xe667d48618e71c2a02e4a1b66ed9def1426938b60000000000000000000004d7](https://kovan.etherscan.io/address/0xba12222222228d8ba445958a75a0704d566bf2c8#readContract)):
* 1,010,529 USDT
* 10,133 aUSDT

The pool page on the base branch shows these same amounts as if they were the balances of USDT and aUSDT held by the bb-a-USD pool. But since the pool only holds 99% of the bb-a-USDT, then it only holds 99% of those values.

**Comment by evgenyboxer** [comment]
@markusbkoch thanks for the clarification.

I've now implemented the suggested changes into a new PR which also handles the aToken prices.

Here is the PR - https://github.com/balancer-labs/frontend-v2/pull/1206

**Pull request closed without merging**